Plugin/Sales/Model/Order/ItemRepository.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Taxjar_SalesTax
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Taxjar
+ * @package    Taxjar_SalesTax
+ * @copyright  Copyright (c) 2020 TaxJar. TaxJar is a trademark of TPS Unlimited, Inc. (http://www.taxjar.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+
+namespace Taxjar\SalesTax\Plugin\Sales\Model\Order;
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+
+class ItemRepository
+{
+    /**
+     * @var ProductRepositoryInterface
+     */
+    protected $productRepository;
+
+    public function __construct(
+        ProductRepositoryInterface $productRepository
+    ) {
+        $this->productRepository = $productRepository;
+    }
+
+    /**
+     * Include the TaxJar product tax code when creating an order item
+     *
+     * @param \Magento\Sales\Model\Order\ItemRepository $subject
+     * @param \Magento\Sales\Api\Data\OrderItemInterface $item
+     */
+    public function beforeSave(
+        \Magento\Sales\Model\Order\ItemRepository $subject,
+        \Magento\Sales\Api\Data\OrderItemInterface $item
+    ) {
+        // Only set the product tax code when first creating the OrderItem
+        if ($item->getItemId() === null) {
+            $product = $this->productRepository->getById($item->getProductId());
```

Comment:
@ryanreeves-taxjar Any potential edge cases where the order item would not have a product before creation? Should we check if `$product` exists to be safe?

Reply:
Deleted products are automatically removed from the quote.  It shouldn't be possible to reach this step with a product that was deleted.  

I'll go ahead and add a catch for NoSuchEntityExceptions anyways.